### PR TITLE
842 explain private subnet block

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

File: pkg/ibmvpc/analysis_output_test.go [hunk 1]
```diff
@@ -460,6 +460,7 @@ var tests = []*commonvpc.VpcGeneralTest{
 		NoLbAbstract: true,
 		Format:       vpcmodel.HTML,
 	},
+	// todo: before merge - this fails sometimes in git. understand why and have fixed (seems in Haim's areas).
 	{
```
Comment (haim-kermany) [inline pkg/ibmvpc/analysis_output_test.go:464]: its not relevant to this PR, I saw it one also on another PR.
better remove this comment, and open an issue

File: pkg/awsvpc/explainability_test.go [hunk 2]
```diff
@@ -23,6 +24,33 @@ var explainTests = []*commonvpc.VpcGeneralTest{
...
+		InputConfig:   "aws_sg_1",
+		ESrc:          "10.240.10.42",
+		EDst:          "161.26.0.0",
+		Format:        vpcmodel.Text,
+		DetailExplain: true,
+	},
+	// connection to the public internet blocked only by private subnet
+	{
+		Name:          "to_external_blocked_only_private_subnet",
+		InputConfig:   "aws_sg_1_allow_sg_private",
```
Comment (haim-kermany) [inline pkg/awsvpc/explainability_test.go:48]: config files should be added to the tests only from the teraform example repository.
and should not be edited.
if you need an specific example, please ask Karen
also notice that we also have aws_acl_1 and aws_mixed examples, to use

**Review by haim-kermany — CHANGES_REQUESTED** [review]

File: pkg/awsvpc/explainability_test.go (hunk 2 above)
Comment (ShiriMoran) [reply pkg/awsvpc/explainability_test.go:48]: We have added manual modified examples in the past. 
However, I opened an issue https://github.com/np-guard/vpc-network-config-analyzer/issues/846 for Karen. Once she generate the required example I will delete mine.
In the meantime lets not have it a bottleneck (Karen said she'll probably get to it Sunday), and have this PR merged.

File: pkg/ibmvpc/analysis_output_test.go (hunk 1 above)
Comment (ShiriMoran) [reply pkg/ibmvpc/analysis_output_test.go]: ok. #847

**Review by haim-kermany — DISMISSED** [review]

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/commonvpc/vpc.go [hunk 3]
```diff
@@ -227,22 +227,32 @@ func (psr *privateSubnetRule) Deny(isIngress bool) bool {
...
+	case detail && psr.subnet.IsPrivate() && !psr.isIngress:
+		return outbound + denyDetailed
+	case detail && psr.subnet.IsPrivate() && psr.isIngress:
+		return inbound + denyDetailed
+	case detail && !psr.subnet.IsPrivate() && !psr.isIngress:
+		return outbound + allowDetailed
+	case detail && !psr.subnet.IsPrivate() && psr.isIngress:
+		return inbound + allowDetailed
+	default: // will never get here
+		return ""
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/commonvpc/vpc.go]: done

File: pkg/awsvpc/examples/out/explain_out/to_external_public_subnet_all_vpcs_explain_detail.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	proxy[10.240.10.42] -> security group GroupId:35 -> network ACL NetworkAclId:65 -> subnet edge -> 
+	InternetGateway internet_gw -> 
+	Public Internet 161.26.0.0/32
+
+
+Details:
+~~~~~~~~
+Path is enabled; The relevant rules are:
+	Egress:
+		external outbound connection is allowed since subnet edge is public
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/awsvpc/examples/out/explain_out/to_external_public_subnet_all_vpcs_explain_detail.txt]: Done
I see I need to add an ingress with external example

**Review by zivnevo — APPROVED** [review]

**Pull request merged**